### PR TITLE
Adds in a widget showcase

### LINKED ISSUE #248: Migrate showcase to examples
**Enhancement**

We should migrate the [showcase](https://github.com/dojo/widgets/tree/1d56be8d649809341af172765505a252b5e3c94c/support/showcase) from `dojo/widgets` do `dojo/examples` and rename it `widget-showcase`.

It should be incorporated in the github pages for this package as well, so it is _live_ when we commit to master.

After that we will need to improve its design via #247.

### PR DESCRIPTION
**Pull request by umaar**
Adds in a widget showcase from https://github.com/dojo/widgets/tree/1d56be8d649809341af172765505a252b5e3c94c/support/showcase and instructs travis-ci to build it and deploy to the gh-pages branch.

Resolves #248

### CODE REVIEW

File: widget-showcase/.gitignore [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/html-report
+/node_modules
+/typings
+.baseDir.ts
+.tscache
+.tsconfig*.json
+coverage-unmapped.json
+npm-debug.log
+output/*
+package-lock.json
```
Comment (agubler) [inline widget-showcase/.gitignore]: nit: missing new line at end of file

File: widget-showcase/src/main.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { ProjectorMixin } from '@dojo/widget-core/mixins/Projector';
+import App from './App';
+import dojoTheme from '@dojo/widgets/themes/dojo/theme';
+
+const root = document.querySelector('showcase') || undefined;
```
Comment (agubler) [inline widget-showcase/src/main.ts]: Let's remove the query selector and just do `projector.append()`, which will append to the body. Also means we can remove `<showcase></showcase>` from the index.html

File: widget-showcase/tests/unit/all.ts [hunk 3]
```diff
@@ -0,0 +1,639 @@
+import * as registerSuite from 'intern!object';
```
Comment (agubler) [inline widget-showcase/tests/unit/all.ts]: This should be `app.ts` and the `all.ts` file should just have `import './app';`

File: widget-showcase/tests/unit/all.ts [hunk 4]
```diff
@@ -0,0 +1,639 @@
...
+		});
+
+		// widget.callListener('onRequestClose', {
+		// 	key: 'dialog'
+		// });
+
+		widget.expectRender(expectedWidget, 'the dialog opens');
+		widget.destroy();
+	}
+});
```
Comment (agubler) [inline widget-showcase/tests/unit/all.ts]: nit: new line at end of file

Comment (umaar) [reply widget-showcase/tests/unit/all.ts]: Done! We should use linting for this one!

File: widget-showcase/tests/unit/all.ts (hunk 3 above)
Comment (umaar) [reply widget-showcase/tests/unit/all.ts]: Done 👍

File: widget-showcase/src/main.ts (hunk 2 above)
Comment (umaar) [reply widget-showcase/src/main.ts]: Ok done. Personally I find it a bit confusing to see an empty HTML file and it's unclear what `projector.append()` does exactly!

File: widget-showcase/tests/unit/app.ts [hunk 5]
```diff
@@ -0,0 +1,789 @@
...
+
+		assignProperties(findKey(expectedWidget, 'split-pane')!, {
+			size: 5
+		});
+
+		widget.expectRender(expectedWidget, 'The split pane size is set');
+
+		/*
+
+		TODO: How to test a nested split pane?
```
Comment (umaar) [inline widget-showcase/tests/unit/app.ts]: @agubler could you help with this TODO please? How do you think I should test this? https://github.com/dojo/examples/blob/0291f47482fb24616bd0d229eef313a9ec787a24/widget-showcase/src/App.ts#L356-L364

**Comment by umaar** [comment]
Please give another review, thanks @agubler ! There's also one thing I need some help with, one of the widgets has a property, that property value is itself another widget! Wondering what the best way of testing that is

**Pull request merged**